### ISSUE TITLE
`flutter run` argument `-t` fails when using : or =

### ISSUE THREAD

**Issue by DanTup** [flutter/flutter#10353]
Normally when passing arguments to processes (the Dart VM in this case) I've been using `--name=value`, eg:

```ts
appArgs.push("--enable-vm-service=0");
appArgs.push("--pause_isolates_on_start=true");
```

However if I try to do this for `flutter run`'s `-t` argument, it fails:

```ts
appArgs.push(`-t=lib/main.dart`);
```

> Target file "=lib/main" not found.

Same with colon:

> Target file ":lib/main.dart" not found.

In fact, it even happens with a space when using node to spawn the process:

```ts
appArgs.push(`-t lib/main.dart`);
```

> Target file " lib\main.dart" not found.

The only way to handle this feels wonky:

```ts
appArgs.push("-t")
appArgs.push("lib/main.dart");
```

Given how close Dart and Flutter are, it would be nice if these were consistent :-)

**Comment by eseidelGoogle** [comment]
The -t part turns out not to be required.  `flutter run` also accepts raw args as main.dart.

Thanks for the report all the same.  FYI @tvolkert

**Comment by DanTup** [comment]
@eseidelGoogle Could the help text be updated accordingly?

Also, I think it'd be useful if the other switches gave examples of how to use them, eg.:

Dart VM:

```text
 --enable-vm-service[=<port>[/<bind-address>]]
```

Flutter:

```text
--diagnostic-port        Listen to the given port for a diagnostic connection.
                         Specifying port 0 will find a random free port.
                         Defaults to the first available port after 8101.
```

There's no indication of whether to use space, equals, colon in the flutter texts.

**Comment by eseidelGoogle** [comment]
Honestly `flutter` is just a dart script, we use package:args for all our args handling. :)  So I suspect this bug is likely going to end up in pakage:args. :)  For help-text fixes, that we certainly can do.  And of course we appreciate you filing (regardless of where the eventual cod change(s) will land).

**Comment by tvolkert** [comment]
We could add `valueHelp` attributes to our options when we call `ArgParser.addOption()`

**Comment by DanTup** [comment]
Yeah, I figured it might be for args, but it also seemed inconsistent (eg. the observatory port didn't fail when I used spaces, colons or equals - though I wasn't able to check whether it actually worked). I presume the Dart VM isn't using args but it still seems like there's value in them being consistent - it'd be nice if all command line apps were but if two close teams at the same company can't do it, what chance does that have? :D
